pkg/build/controller/controller.go

```diff
@@ -125,29 +125,8 @@ func (bc *BuildController) nextBuildPhase(build *buildapi.Build) error {
 	// Set the build phase, which will be persisted if no error occurs.
 	build.Status.Phase = buildapi.BuildPhasePending
 
-	// Make a copy to avoid mutating the build from this point on.
-	copy, err := kapi.Scheme.Copy(build)
-	if err != nil {
-		return fmt.Errorf("unable to copy build: %v", err)
-	}
-	buildCopy := copy.(*buildapi.Build)
-
-	// TODO(rhcarvalho)
-	// The S2I and Docker builders expect build.Spec.Output.To to contain a
-	// resolved reference to a Docker image. Since build.Spec is immutable, we
-	// change a copy (that is never persisted) and pass it to
-	// bc.BuildStrategy.CreateBuildPod. We should make the builders use
-	// build.Status.OutputDockerImageReference, what will make copying the build
-	// unnecessary.
-	if build.Spec.Output.To != nil && len(build.Spec.Output.To.Name) != 0 {
-		buildCopy.Spec.Output.To = &kapi.ObjectReference{
-			Kind: "DockerImage",
-			Name: ref,
-		}
-	}
```

Comment:
@rhcarvalho @PI-Victor we can't stop setting bulid.Spec.Output.To, or you'll potentially break existing custom build images that assume the value is set correctly.